flask_ask/core.py

```diff
@@ -869,6 +870,12 @@ def get_source(self, environment, template):
         if self.last_mtime != os.path.getmtime(self.path):
             self._reload_mapping()
         if template in self.mapping:
-            source = self.mapping[template]
+            # convert to template list if not already
+            # choose one at random
+            options = self.mapping[template]
+            if not isinstance(options, list):
+                options = [options]
+
+            source = random.choice(options)
```

Comment:
Very much needed feature IMHO. I merged it in my own fork, and I'm using it now.
I believe for correctness, the next line should also be changed IIUC, from:

    return source, None, lambda: source == self.mapping.get(template)

to:

    return source, None, lambda: options == self.mapping.get(template)

because `source` and `self.mapping.get(template)` will otherwise never be equal.

Reply:
I had a sense that it was wrong when writing this patch but it was working as expected so I didn't want to touch it. This will require insight from @johnwheeler

Reply:
Actually, I since modified this very condition to make sure it *never* returns True.
The lambda is used by Jinja2 loader to determine when to reload the templates.
If we do the right thing, given the interface, which would be `options == self.mapping.get(template)` what we end up with is *always* the same template every time (same random sentence always chosen). This openly works against what we're trying to do.

Instead, we should just use `lambda: True`, to signal that we always want Jinja to reload the template, thus picking a random one among the possibilities.

Reply:
and [this updated PR](https://github.com/johnwheeler/flask-ask/pull/241) will handle random templates correctly, or at least it does for me.